src/mrnet/network/reaction_network.py

```diff
@@ -1594,6 +1632,162 @@ def mols_w_cuttoff(RN_pr_solved, cutoff=0, build_pruned_network=True):
+        :param node: node string, ex "1+PR_2,3+4"
+        """
+        react_list = []
+        prod_list = []
+        if "PR" in node and "+" in node.split(",")[1]:
+            reactant = int(node.split("+PR_")[0])
+            pr = int(node.split("+PR_")[1].split(",")[0])
+            product1 = int(node.split("+PR_")[1].split(",")[1].split("+")[0])
+            product2 = int(node.split("+PR_")[1].split(",")[1].split("+")[1])
+            react_list = [reactant, pr]
+            prod_list = [product1, product2]
+            prod_list.sort()
+        elif "PR" in node and "+" not in node.split(",")[1]:
+            reactant = int(node.split("+PR_")[0])
+            pr = int(node.split("+PR_")[1].split(",")[0])
+            product = int(node.split("+PR_")[1].split(",")[1])
+            react_list = [reactant, pr]
+            prod_list = [product]
+        elif "PR" not in node and "+" in node:
+            reactant = int(node.split(",")[0])
+            product1 = int(node.split(",")[1].split("+")[0])
+            product2 = int(node.split(",")[1].split("+")[1])
+            prod_list = [product1, product2]
+            prod_list.sort()
+            react_list = [reactant]
+            prod_list = [product1, product2]
+            prod_list.sort()
+        elif "PR" not in node and "+" not in node:
+            reactant = int(node.split(",")[0])
+            product = int(node.split(",")[1])
+            react_list = [reactant]
+            prod_list = [product]
+        else:
+            print("Something wrong while parsing reaction node string", node)
+
+        return react_list, prod_list
+
+    @staticmethod
+    def identify_concerted_rxns_via_intermediates_NEW(RN, mols_to_keep=None,
+                                                  single_elem_interm_ignore=["C1", "H1", "O1", "Li1"]):
+        """
+            A method to identify concerted reactions by looping through high enery intermediates
+        :param RN: ReactionNetwork atleast built with elementary reaction
+        :param mols_to_keep: List of pruned molecules, if not running then a list of all molecule nodes in the
+        RN_pr_solved
+        :param single_elem_interm_ignore: List of formula of high energy intermediates to ignore
+        :return: list of reactions
+        """
+        print("identify_concerted_rxns_via_intermediates start", time.time())
+        flag = True
+        if mols_to_keep is None:
+            mols_to_keep = list(range(0, len(RN.entries_list)))
+        mols_to_keep.append(None)
+        count_total = 0
+        reactions = []
+        unique_reactions = []
+        not_wanted_formula = single_elem_interm_ignore
+        for entry in RN.entries_list:
+            node = entry.parameters["ind"]
+            if RN.entries_list[node].formula not in not_wanted_formula and RN.graph.nodes[node][
+                "bipartite"] == 0 and node not in RN.not_reachable_nodes and node not in RN.unsolvable_PRs:
```

Comment:
For code clarity purposes, you might want to break this up somehow. Comments would also be helpful